connect/client.go

```diff
@@ -106,6 +109,53 @@ type restClient struct {
 	client    httpClient
 }
 
+func (rs *restClient) GetFiles(itemUUID string, vaultUUID string) ([]onepassword.File, error) {
+	span := rs.tracer.StartSpan("GetFiles")
+	defer span.Finish()
+
+	jsonURL := fmt.Sprintf("/v1/vaults/%s/items/%s/files", vaultUUID, itemUUID)
+	request, err := rs.buildRequest(http.MethodGet, jsonURL, http.NoBody, span)
+	if err != nil {
+		return nil, err
+	}
+	response, err := rs.client.Do(request)
+	if err != nil {
+		return nil, err
+	}
+	if err := expectMinimumConnectVersion(response, version{1, 3, 0}); err != nil {
+		return nil, err
+	}
+	var files []onepassword.File
+	if err := parseResponse(response, http.StatusOK, &files); err != nil {
+		return nil, err
+	}
+
+	return files, nil
+}
+
+func (rs *restClient) DownloadFile(fileUUID, itemUUID, vaultUUID, targetDirectory string) error {
+	file, err := rs.GetFile(fileUUID, itemUUID, vaultUUID)
+	if err != nil {
+		return err
+	}
+	content, err := rs.GetFileContent(file)
+	if err != nil {
+		return err
+	}
```

Comment:
`rs.GetFileContent()` loads the complete file into memory. In most cases this should not be an issue. But in case of a large file, I can see some benefit in letting `DownloadFile()` directly stream the contents to a file. This would mean the function can technically handle files that are bigger than the system's memory. 

This could be achieved by calling `io.Copy()` on the request Body directly instead of having the `ioutil.ReadAll()` in between.

What do you think about this?

Reply:
Great idea! I've just took a shot at implementing it. Let me know what you think.

Reply:
That's exactly what I meant :+1: